internal/provider/packer/resources/bucket/resource_packer_bucket.go

```diff
@@ -0,0 +1,248 @@
+	resp.Diagnostics.AddError("Unexpected provider error", "This is an internal error, please report this issue to the provider developers")
+
+}
+
+func (r *resourcePackerBucket) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*clients.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *resourcePackerBucket) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+	modifiers.ModifyPlanForDefaultProjectChange(ctx, r.client.Config.ProjectID, req.State, req.Config, req.Plan, resp)
+}
+
+type bucket struct {
+	ProjectID      types.String `tfsdk:"project_id"`
+	OrganizationID types.String `tfsdk:"organization_id"`
+	Name           types.String `tfsdk:"name"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+	CreatedAt      types.String `tfsdk:"created_at"`
+}
+
+func (r *resourcePackerBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan bucket
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectID := r.client.Config.ProjectID
+	if !plan.ProjectID.IsUnknown() {
+		projectID = plan.ProjectID.ValueString()
+	}
+
+	orgID := r.client.Config.OrganizationID
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: orgID,
+		ProjectID:      projectID,
+	}
+	name := plan.Name.ValueString()
+	res, err := packerv2.CreateBucket(ctx, r.client, loc, name)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating bucket", err.Error())
+		return
+	}
+
+	plan.ResourceName = types.StringValue(res.ResourceName)
+	plan.ProjectID = types.StringValue(res.Location.ProjectID)
+	plan.OrganizationID = types.StringValue(res.Location.OrganizationID)
+	plan.CreatedAt = types.StringValue(res.CreatedAt.String())
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourcePackerBucket) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state bucket
+
+	// Read Terraform prior state into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	params := packerservice.NewPackerServiceGetBucketParams()
+	params.SetLocationOrganizationID(state.OrganizationID.ValueString())
+
+	params.SetBucketName(state.Name.ValueString())
+	bucketResp, err := r.client.PackerV2.PackerServiceGetBucket(params, nil)
+
+	params.SetLocationProjectID(state.ProjectID.ValueString())
```

Comment:
Why is the location set to `params` after the request? 🤔

Reply:
Did this work?

Reply:
OMG good catch

So because I had the env variable `HCP_PROJECT_ID` set this was working, I think `hcp-sdk-go` is picking up the HCP_PROJECT_ID and setting it by default to the request when empty, which is a bit confusing

When I unset my `HCP_PROJECT_ID` and put the project ID in the template it fails, fixing this now

Reply:
```
➜  tmp git:(hpr-1729-bucket-resource) ✗ env | grep HCP_PROJECT_ID
➜  tmp git:(hpr-1729-bucket-resource) ✗ terraform import hcp_packer_bucket.example packer/project/f6f70e9e-7c95-4eed-aa49-d9d3f581622f/bucket/bwehh
hcp_packer_bucket.example: Importing from ID "packer/project/redacted/bucket/bwehh"...
hcp_packer_bucket.example: Import prepared!
  Prepared hcp_packer_bucket for import
hcp_packer_bucket.example: Refreshing state... [name=bwehh]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

➜  tmp git:(hpr-1729-bucket-resource) ✗ cat main.tf
terraform {
  required_providers {
    hcp = {
      source  = "localhost/providers/hcp"
      version = "0.0.1"
    }
  }
}

resource "hcp_packer_bucket" "example" {
        name = "bwehh"
        project_id = "redacted"
}
```

It works as expected now regardless of env variables, great catch

Reply:
Awesome! Thank you